core/src/main/scala/dagr/core/reporting/ExecutionLogger.scala

```diff
@@ -0,0 +1,211 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 LLC Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.core.reporting
+
+import com.fulcrumgenomics.commons.CommonsDef.{FilePath, SafelyClosable}
+import com.fulcrumgenomics.commons.io.Io
+import dagr.core.reporting.ExecutionLogger.{Definition, Relationship, Status}
+import dagr.core.reporting.ReportingDef._
+import dagr.core.tasksystem.Task
+import dagr.api.models.TaskStatus
+import dagr.core.tasksystem.Task.TaskInfoLike
+
+import scala.collection.mutable
+
+object ExecutionLogger {
```

Comment:
This is the class that logs the execution state and relationships between tasks.  The file produced is used as input to the replay system.

Reply:
Maybe we could call it `ReplayLogger` then?